internal/service/cloudwatch/contributor_insight_rule.go

```diff
@@ -109,6 +108,26 @@ func (r *resourceContributorInsightRule) Create(ctx context.Context, req resourc
 	cirARN := r.Meta().RegionalARN(ctx, "cloudwatch", fmt.Sprintf("insight-rule/%s", plan.RuleName.ValueString()))
 	plan.ResourceARN = fwflex.StringValueToFramework(ctx, cirARN)
 
+	if !plan.RuleState.IsNull() {
+		if plan.RuleState.ValueString() == "ENABLED" {
+			_, err = conn.EnableInsightRules(ctx, &cloudwatch.EnableInsightRulesInput{
+				RuleNames: []string{plan.RuleName.ValueString()},
+			})
+		} else if plan.RuleState.ValueString() == "DISABLED" {
+			_, err = conn.DisableInsightRules(ctx, &cloudwatch.DisableInsightRulesInput{
+				RuleNames: []string{plan.RuleName.ValueString()},
+			})
+		}
```

Comment:
is the default state of a newly created rule `ENABLED` or `DISABLED`? If it is disabled by default then the only possible operation during this create would be `EnableInsightRule`. You can probably simplify this by only calling the operation that is possible

Reply:
Might also be worth making `ENABLED` and `DISABLED` constants since they are referenced often